runtime/src/main/java/com/datastax/oss/quarkus/runtime/driver/QuarkusNettyOptions.java

```diff
@@ -0,0 +1,164 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.quarkus.runtime.driver;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.internal.core.context.NettyOptions;
+import com.datastax.oss.driver.internal.core.util.concurrent.BlockingOperation;
+import com.datastax.oss.driver.shaded.guava.common.util.concurrent.ThreadFactoryBuilder;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.FixedRecvByteBufAllocator;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.util.HashedWheelTimer;
+import io.netty.util.Timer;
+import io.netty.util.concurrent.DefaultPromise;
+import io.netty.util.concurrent.EventExecutorGroup;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GlobalEventExecutor;
+import io.netty.util.internal.PlatformDependent;
+import java.time.Duration;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QuarkusNettyOptions implements NettyOptions {
+  private static final Logger LOG = LoggerFactory.getLogger(QuarkusNettyOptions.class);
+
+  private final DriverExecutionProfile config;
+  private final EventLoopGroup ioEventLoopGroup;
+  private final EventLoopGroup adminEventLoopGroup;
+  private final Timer timer;
+
+  public QuarkusNettyOptions(
+      QuarkusDriverContext context,
+      EventLoopGroup ioEventLoopGroup,
+      EventLoopGroup adminEventLoopGroup) {
+    this.config = context.getConfig().getDefaultProfile();
+    boolean daemon = config.getBoolean(DefaultDriverOption.NETTY_DAEMON);
+    // the NETTY_IO_SHUTDOWN_QUIET_PERIOD, NETTY_IO_SHUTDOWN_TIMEOUT, NETTY_IO_SHUTDOWN_UNIT,
+    // NETTY_ADMIN_SHUTDOWN_QUIET_PERIOD,
+    // NETTY_ADMIN_SHUTDOWN_TIMEOUT, NETTY_ADMIN_SHUTDOWN_UNIT
+    // are ignored when using quarkus netty even loops
+
+    ThreadFactory safeFactory = new BlockingOperation.SafeThreadFactory();
+
+    this.ioEventLoopGroup = ioEventLoopGroup;
+    this.adminEventLoopGroup = adminEventLoopGroup;
+
+    // setup the Timer
+    ThreadFactory timerThreadFactory =
+        new ThreadFactoryBuilder()
+            .setThreadFactory(safeFactory)
+            .setNameFormat(context.getSessionName() + "-timer-%d")
+            .setDaemon(daemon)
+            .build();
+
+    Duration tickDuration = config.getDuration(DefaultDriverOption.NETTY_TIMER_TICK_DURATION);
+    // JAVA-2264: tick durations on Windows cannot be less than 100 milliseconds,
+    // see https://github.com/netty/netty/issues/356.
+    if (PlatformDependent.isWindows() && tickDuration.toMillis() < 100) {
```

Comment:
A bit sad the we need to repeat this code here.

Reply:
Agree, there is some copy-paste here form `DefaultNettyOptions`.
We could refactor the original `DefaultNettyOptions` to:
```
  // setup the Timer
    ThreadFactory timerThreadFactory =
        new ThreadFactoryBuilder()
            .setThreadFactory(safeFactory)
            .setNameFormat(context.getSessionName() + "-timer-%d")
            .setDaemon(daemon)
            .build();

    this.timer = constructTimer(timerThreadFactory);
  }

  protected Timer constructTimer(ThreadFactory timerThreadFactory)
  {
    Duration tickDuration = config.getDuration(DefaultDriverOption.NETTY_TIMER_TICK_DURATION);
    // JAVA-2264: tick durations on Windows cannot be less than 100 milliseconds,
    // see https://github.com/netty/netty/issues/356.
    if (PlatformDependent.isWindows() && tickDuration.toMillis() < 100) {
      LOG.warn(
          "Timer tick duration was set to a value too aggressive for Windows: {} ms; "
              + "doing so is known to cause extreme CPU usage. "
              + "Please set advanced.netty.timer.tick-duration to 100 ms or higher.",
          tickDuration.toMillis());
    }
    return 
        new HashedWheelTimer(
            timerThreadFactory,
            tickDuration.toNanos(),
            TimeUnit.NANOSECONDS,
            config.getInt(DefaultDriverOption.NETTY_TIMER_TICKS_PER_WHEEL));
  }
```
and use the `constructTimer()` method here. wdyt?

Reply:
Well you said you prefer to keep the copy-paste in sync so 🤷‍♂️ 
In any case, inheriting from `DefaultNettyOptions` is tricky because all the stuff that happens in its constructor. A better solution imo would be to isolate the Timer creation into a factory class.